versionedRouter.js

```diff
@@ -915,7 +915,7 @@ async function handleProxyRequest(destination, ctx) {
       destination
     });
     response = destNetworkHandler.responseHandler(
-      processedProxyResponse,
+      { ...processedProxyResponse, metadata },
```

Comment:
To avoid collisions, we should not merge metadata into the proxy response.

Reply:
As it is internal, we don't have to worry about backward compatibility.

Reply:
You can send `metadata` (`reqMetadata`) as a third optional argument to this function.

Reply:
Why are you saying there will be collisions ?

Reply:
We can't be sure that `processedProxyResponse` would not contain a field named `metadata`. It's a preemptive measure.

Reply:
Then we can put another named key `rudderJobMetadata` instead of `metadata` and proceed with the same implementation(as in the changes)
```javascript
response = destNetworkHandler.responseHandler(
      processedProxyResponse,
      { ...processedProxyResponse, rudderJobMetadata: metadata },
      ...);
```